csrc/scheduler/normalization_inner_outer_tma_ws.cpp

```diff
@@ -641,7 +559,54 @@ void scheduleFusion(Fusion* fusion, const ReductionParams* rparams) {
   for (auto output : dummy_outputs) {
     fusion->removeOutput(output);
   }
-  inlineMost();
+  // inline
+  if (rparams->circular_buffer_options.isEnable()) {
+    std::unordered_map<TensorView*, int64_t> tv_inline_pos_map;
+    // TMA loaded tv may have a domain of either
+    // [I/Unroll/BIDy, BIDy, Unroll | Bulk]
+    // or
+    // [I/BIDy, BIDy, | Bulk]
+    // or
+    // [Bulk]
+    // Set inline position after BIDy, so all the unrolled TMA loads
+    // share the same barrier.
+    int64_t tma_inline_pos = 2;
+    for (auto tv : tma_load_tvs) {
+      if (tv->nDims() >= tma_inline_pos + 1) {
+        tv_inline_pos_map.emplace(tv, tma_inline_pos);
+      }
+    }
+    // For smem consumers, set inline position to the same as tma load tvs.
+    // This allows quick release the shared memory barrier to launch the
+    // next TMA load. Otherwise, the inline position is after Unroll axis.
```

Comment:
I think you meant `before Unroll axis` instead?
```suggestion
    // next TMA load. Otherwise, the inline position is before Unroll axis.
```